src/controllers/DoctorController.java

```diff
@@ -68,63 +68,23 @@ private Command LoadPatient() {
         };
     }
 
-/* PENDING IMPLEMENTATION IN PHASE 2
-    private Command ViewSchedule(){
+    private Command ViewAppointments() {
         return (x) -> {
-            LocalDate viewDate = doctorScreenView.viewSchedulePrompt();
-            doctorScreenView.viewAppointments(new AppointmentManager(getDatabase()).getScheduleData(doctorData, viewDate));
+            ArrayList<AppointmentData> appointments = appointmentManager.getDoctorAppointments(doctorData);
+            doctorScreenView.viewAppointments(appointments);
         };
     }
 
-    private Command ViewAllDoctorAppointments(){
-        return (x) -> doctorScreenView.viewAppointments(new AppointmentManager(getDatabase())
-                .getDoctorAppointments(doctorData));
-    }
-
-    private Command ViewAllAppointments(){
-        return (x) -> doctorScreenView.viewAppointments(new AppointmentManager(getDatabase()).getAllAppointments());
-
-    }
 
-    private Command newAvailability() {
+    private Command ViewSchedule() {
         return (x) -> {
-            ArrayList<Integer> availabilityInfo = doctorScreenView.addAvailabilityPrompt();
-            new AppointmentManager(getDatabase()).newAvailability(doctorData, DayOfWeek.of(availabilityInfo.get(0)),
-                    availabilityInfo.get(1), availabilityInfo.get(2), availabilityInfo.get(3));
-        };
-    }
-
-    private Command deleteAvailability() {
-        return (x) -> {
-            Integer deleteInteger = doctorScreenView.deleteAvailabilityPrompt(new ContactManager(getDatabase())
-                    .getContactData(doctorData), new AppointmentManager(getDatabase())
-                    .getAvailabilityData(doctorData));
-            ArrayList<AvailabilityData> availability = doctorData.getAvailability();
-            if (deleteInteger >= 0 & deleteInteger < availability.size()) {
-                new AppointmentManager(getDatabase()).removeAvailability(doctorData,
-                        doctorData.getAvailability().get(deleteInteger));
-            }
+            LocalDate viewDate = doctorScreenView.viewSchedulePrompt();
+            doctorScreenView.viewAppointments(new AppointmentManager(getDatabase()).getSingleDayAppointment(doctorData, viewDate));
         };
     }
 
-    private Command deleteAbsence() {
-        return (x) -> {
-            Integer deleteInteger = doctorScreenView.deleteAbsencePrompt(new ContactManager(getDatabase())
-                    .getContactData(doctorData), doctorData.getAbsence().stream()
-                    .map(TimeBlockData::new)
-                    .collect(Collectors.toCollection(ArrayList::new)));
-            new AppointmentManager(getDatabase()).deleteAbsence(doctorData, doctorData.getAbsence().get(deleteInteger));
-        };
+    private Command ViewAllDoctorAppointments() {
+        return (x) -> doctorScreenView.viewAppointments(new AppointmentManager(getDatabase())
+                .getDoctorAppointments(doctorData));
     }
```

Comment:
for all three view methods from line 71 to 89 a check and no appointments message should be added.